### ISSUE TITLE
Is it possible to listen on multiple actions?

### ISSUE THREAD

**Issue by jmyrland** [ctrlplusb/easy-peasy#205]
Loving the changes so far & the updated docs! 👍 I'm only sad to see `select`s going away (or replaced by `selector`) 😢 

As the title specifies, is there a quick way to apply the same action/thunk for multiple "events"?

e.g.
```ts
const model = { 
  //...
  onUpdate1: thunk(
    (actions, payload, { getStoreState }) => {
      const currentState = getStoreState();
      const someData = calculateNewState(currentState);
      actions.update(someData);
    },
    { listenTo: someOtherModel.action1 } // this..
  ),
  onUpdate2: thunk(
    (actions, payload, { getStoreState }) => {
      const currentState = getStoreState();
      const someData = calculateNewState(currentState);
      actions.update(someData);
    },
    { listenTo: someOtherModel.action2 } // is..
  ),
  onUpdate3: thunk(
    (actions, payload, { getStoreState }) => {
      const currentState = getStoreState();
      const someData = calculateNewState(currentState);
      actions.update(someData);
    },
    { listenTo: someOtherModel.action3 }  // tedious..
  ),
  //...
}
```

> In this case I'm not really interested in the incoming `payload`, but I'm instead relying on `getStoreState `

I figure I can do something like this to shorten it:

```ts
const applyUpdates = (actions, payload, { getStoreState }) => {
  const currentState = getStoreState();
  const someData = calculateNewState(currentState);
  actions.update(someData);
};
const model = { 
  //...
  onUpdate1: thunk(
    applyUpdates,
    { listenTo: someOtherModel.action1 }
  ),
  onUpdate2: thunk(
    applyUpdates ,
    { listenTo: someOtherModel.action2 }
  ),
  onUpdate3: thunk(
    applyUpdates ,
    { listenTo: someOtherModel.action3 }
  ),
  //...
}
```

But what I ideally want to do, is something like this:
```ts
const model = { 
  //...
  onUpdate: thunk(
    (actions, payload, { getStoreState }) => {
      const currentState = getStoreState();
      const someData = calculateNewState(currentState);
      actions.update(someData);
    },
    { listenTo: [someOtherModel.action1, someOtherModel.action2, someOtherModel.action3] }
  ),
}
```
Is there a way to achieve the above solution - or something similar?

**Comment by ctrlplusb** [comment]
Hey @jmyrland 

Glad to hear you are overall liking the direction of things. 

In regards to `select` being deprecated - I'm really sorry about this one. I really preferred the `select` API myself, but it was too much of a performance footgun as it was really hard for me to optimise them running without resorting to very exotic ideas such as Proxy based property access monitoring - which apart from significantly increasing the package may introduce some really hard to debug issues.

I'm really happy to hear you make this request in regards to being able to listen to an array of actions - this is something I had already anticipated. It makes me feel good to hear an actual request come out though.

We'll definitely extend the API to allow for this case. 👍

**Comment by ctrlplusb** [comment]
Released in v2.5.0 👍

Simply set an array of targets on `listenTo`. I'll publish updated website soon.

**Comment by jmyrland** [comment]
Amazing stuff @ctrlplusb 👍 Will check it out tomorrow 😁 

I see your point regarding `select`. Would [getters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get) be an option for the `selectors`? By that I mean that the function invocation is "hidden" behind the getter property.

(Maybe that is what you meant by "Proxy based property access monitoring"?)

On another note, with the change, it's now easier to identify "selectors". And the change (I assume) is better suited to expose "parameterized getter" functions (like. `getById(id: string)`).

To me, it feels more "clean" (in lack of better terms 🤔) to just access a property of the state, rather than calling a method. Just some food for thought 😄
